src/query/api/v1/handler/graphite/render.go

```diff
@@ -164,8 +173,9 @@ func (h *renderHandler) serveHTTP(
 			}
 
 			mu.Lock()
-			results[i] = targetSeries
+			meta = meta.CombineMetadata(targetSeries.Metadata)
 			mu.Unlock()
+			results[i] = targetSeries
```

Comment:
Should the `results[i] = targetSeries` remain inside the mu.Lock() and mu.Unlock()?  I guess it's inserting into unique index so doesn't necessarily need to be...

Reply:
I think it's the latter, but moved it into the lock just to be safe; doubt there are any perf implications here